src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -317,7 +317,7 @@ private AddTxResult AddCore(Transaction tx, bool isPersistentBroadcast, bool isR
                     _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, cost overflow.");
                 return AddTxResult.BalanceOverflow;
             }
-            else if ((account?.Balance ?? UInt256.Zero) < cost)
+            else if ((account?.Balance ?? UInt256.Zero) < cost && !tx.IsEip1559)
```

Comment:
I would like to understand this change in details

Reply:
That was a temporary ugly hack. Changed to calculate the balance properly in the context of 1559. However, I will add a few more checks to TxPool.